lib/charms/tempo_k8s/v2/tracing.py

```diff
@@ -595,12 +614,15 @@ def publish_receivers(self, receivers: Sequence[RawReceiver]):
         for relation in self.relations:
```

Comment:
@PietroPasotti @mmkay that would would update all relations app data with "recievers" they didn't necessarily subscribe to. Is this an intended behavior?  I don't think that is correct

Reply:
If I recall right I had the same discussion with @PietroPasotti before and I think the result was that Tempo should expose any receiver they have. Correct me if I'm wrong @PietroPasotti  :)

Reply:
Observation:
When a tracing relation is removed, the endpoints/protocols it subscribed to, still exists in the databag of *other* charm relations

Reply:
that's weird. We're probably missing a relation-removed observer